core/dnsserver/onstartup.go

```diff
@@ -2,24 +2,39 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
+
+	"github.com/coredns/coredns/plugin/pkg/dnsutil"
 )
 
+// checkZoneSyntax() checks whether the given string match 1035 Preferred Syntax or not.
+func checkZoneSyntax(zone string) bool {
+	if(zone == "." || dnsutil.IsReverse(zone) != 0){
+		return true
+	}
+	regex1035PreferredSyntax, _ := regexp.MatchString(`^([A-Za-z]([A-Za-z0-9-]*[A-Za-z0-9])?(\.[A-Za-z]([A-Za-z0-9-]*[A-Za-z0-9])?)*\.)$`, zone)
```

Comment:
Your regex will incorrectly match a zone starting with `.` such as `.example.`.
Please use the regex I have suggested. i.e.

```suggestion
	regex1035PreferredSyntax, _ := regexp.MatchString(`^([A-Za-z]([A-Za-z0-9-]*[A-Za-z0-9])?\.)+$`, zone)
```